tools/monorepo-utils/src/ci-jobs/lib/job-processing.ts

```diff
@@ -70,17 +74,48 @@ function replaceCommandVars( command: string, options: CreateOptions ): string {
 	} );
 }
 
+/**
+ * Multiplies a job based on the shards job config. It updates the job names and command - currently only supporting Playwright sharding.
+ *
+ * @param {TestJob}       job       The job to be multiplied.
+ * @param {TestJobConfig} jobConfig The job config.
+ * @return {TestJob[]} The list of sharded jobs.
+ */
+export function getShardedJobs(
+	job: TestJob,
+	jobConfig: TestJobConfig
+): TestJob[] {
+	let createdJobs = [];
+	if ( jobConfig.shards <= 1 ) {
+		createdJobs.push( job );
+	} else {
+		createdJobs = Array( jobConfig.shards )
+			.fill( null )
+			.map( ( _, i ) => {
+				const jobCopy = JSON.parse( JSON.stringify( job ) );
+				jobCopy.shardNumber = i + 1;
+				jobCopy.name = `${ job.name } ${ jobCopy.shardNumber }/${ jobConfig.shards }`;
+				jobCopy.command = `${ job.command } --shard ${ jobCopy.shardNumber }/${ jobConfig.shards }`;
```

Comment:
One caveat to this approach is that it limits sharding to _only_ tools that match Playwright's `--shard` option.

Reply:
How would you feel about supporting sharding using the current configuration object structure? You would add shards by just adding new test job entries with a command that includes `--shard x/x` and a name that reflects the shard. It's a bit more verbose in the config file but it's also more explicit, and really, it won't ever change once you've set it.

Reply:
> One caveat to this approach is that it limits sharding to only tools that match Playwright's --shard option.

> How would you feel about supporting sharding using the current configuration object structure? You would add shards by just adding new test job entries with a command that includes --shard x/x and a name that reflects the shard. It's a bit more verbose in the config file but it's also more explicit, and really, it won't ever change once you've set it.

Yes, that is so much simpler, I was ready to revert the sharding logic, but thinking about it some more I'm not sure anymore. I'm not so sure about the `won't ever change once you've set it` part to be honest. New tests are constantly being added, suites are maintained, and more events will need to be supported with some of the projects probably defined per trigger (e.g. for the release event we'll want a different test suite for multiple environments, each of them sharded). This will lead to a very long list of defined projects making is harder to navigate and understand what's being defined, and making it easy to miss stuff. 
Another example would be the blocks e2e, [currently being split into 20(!)](https://github.com/woocommerce/woocommerce/actions/runs/8265811969?pr=45435) jobs. I would not want to have 20 projects defined TBH.
If it's ok with you, I would keep it like this for now and think about it some more. We can improve in a future iteration, once we include more tests and better understand our needs.

Reply:
The other idea I had when initially reviewing this PR was that we could add an alternative option to `"command"` for tests. Something like `"shardCommands"`:

```json
{
	"ci": {
		"tests": [
			{
				"name": "E2E",
				"shardCommands": [
					"test:php:env --shard=1/2",
					"test:php:env --shard=2/2"
				],
				...
			}
		]
	}
}
```

This would:

* Error when parsing config if both `"command"` and `"shardCommands"` are set.
* Suffix the name with the shard index + 1 (`E2E - Shard 1`, `E2E - Shard 2`, etc)
* Create jobs for each `"shardCommands"` entry with the same options for the rest of the test job.

It supports both Playwright sharding _and_ other kinds of sharding like PHPUnit groups and the like. The config stays very clean and the support for sharding becomes very explicit.

> If it's ok with you, I would keep it like this for now and think about it some more. We can improve in a future iteration, once we include more tests and better understand our needs.

Anything can be changed later :smile:

Reply:
It's also still compatible with a `"reporter"` config if you do that later since we can refer to the `"shardCommand"` index!

Reply:
Updated now, I chose the third option, to configure `shardingArguments` as an array, and appended to the command.